SONiC_QueryParameterForGet_HLD.md

```diff
@@ -0,0 +1,454 @@
+
+**Field Support:**    (RFC 8040)4.8.3. The "fields"Query Parameter
+
+The"fields" query parameter is used to optionally identify data nodes within the target resource to be retrieved in a GET method. The client can use this parameter to retrieve a subset of all nodes in a resource.
+
+| **Field Type**                           |                                          |                                          |
+| ---------------------------------------- | ---------------------------------------- | ---------------------------------------- |
+| single  field****                        | **Request**     <br />curl -X GET "https://localhost/restconf/data/openconfignetwork-  instance:network-instances/network-instance=default/protocols/protocol=IGMP_SNOOPING,IGMP-SNOOPING/openconfig-network-instance-deviation:igmp-snooping/interfaces/interface**?fields=config/enabled**"  -H "accept: application/yang-data+json" -H "Content-Type:  application/yangdata+json"<br />     **Response:**  {<br />     "openconfig-network-instance- deviation:config": {<br />         "enabled": true<br />     }<br />  } | **Request :**  <br />curl -X GET "https://localhost/restconf/data/openconfig-networkinstance:  network-instances/network-instance=default/protocols  /protocol=IGMP_SNOOPING,IGMP-SNOOPING/openconfig-  network-instance-deviation:igmp-snooping**?content=all**"  -H "accept:application/yang-data+json" -H "Content-Type:  application/yangdata+json"<br />     **Response:**  {<br />     "openconfig-network-instance-   deviation:igmpsnooping":    {<br />               "interfaces": {<br />                   "interface": [{<br />                       "config": {<br />                           "enabled": true,<br />                            "last-memberquery-interval": 2000,                            "name": "Vlan1"<br />                      },<br />                      "name": "Vlan1",<br />                      "state": {<br />                          "enabled": true,<br />                         "last-memberquery-interval": 2000,<br />                         "name": "Vlan1"<br />                      }<br />                  }, {<br />                       "config": {<br />                          "enabled": false,<br />                           "last-memberquery-interval": 4000,<br />         "name": "Vlan2",<br />         "version": 3<br />       },<br />       "name": "Vlan2",<br />       "state": {<br />         "enabled": false,<br />         "last-memberquery-interval":  4000,<br />         "name": "Vlan2",<br />         "version": 3<br />       }<br />    }]<br />              }<br />    }<br />  } |

```

Comment:
If a field belongs to a list, should it retrieve the field from all entries of the lists?

Reply:
Yes, the server returns the selected data nodes from all entries.
e.g. using sub-selector with target resource list
GET /restconf/data/openconfig-interfaces:interfaces/interface?fields=state(name;admin-status;oper-status;counters)